server/server.go

```diff
@@ -76,13 +78,28 @@ func checkServerRunning(bind string) bool {
 
 func StartCredentialsServer(creds *vault.VaultCredentials) error {
 	if !checkServerRunning(metadataBind) {
-		log.Printf("Starting `aws-vault server` as root in the background")
-		cmd := exec.Command("sudo", "-b", os.Args[0], "server")
-		cmd.Stdin = os.Stdin
-		cmd.Stdout = os.Stdout
-		cmd.Stderr = os.Stderr
-		if err := cmd.Run(); err != nil {
-			return err
+		if runtime.GOOS == "windows" {
+			log.Printf("Starting `aws-vault server` in the background")
+			cmd := exec.Command(os.Args[0], "server")
+			cmd.Stdin = os.Stdin
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			if err := cmd.Start(); err != nil {
+				return err
+			}
+			time.Sleep(time.Second * 1)
+			if !checkServerRunning(metadataBind) {
+				return errors.New("The credential proxy server isn't running. Run aws-vault server as Administrator in the background and then try this command again")
+			}
+		} else {
+			log.Printf("Starting `aws-vault server` as root in the background")
+			cmd := exec.Command("sudo", "-b", os.Args[0], "server")
+			cmd.Stdin = os.Stdin
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			if err := cmd.Run(); err != nil {
+				return err
+			}
```

Comment:
Might read a bit nicer if we pull this logic out into 2 functions, `StartCredentialsServer` and `StartCredentialsServerWithSudo`? And might be worth doing the `checkServerRunning` in both cases - there are failure scenarios if sudo doesn't exist for example

Reply:
if sudo doesn't exist then we still get to see the error as we start sudo with [`cmd.Run()`](https://golang.org/pkg/os/exec/#Cmd.Run). This launches sudo and waits for it to finish - which it does as soon as it starts the process in the background `-b`